backend/__tests__/api.test.js

```diff
@@ -9,85 +9,242 @@ let userId;
 let jwt;
 
 beforeAll(async () => {
-    await mongoose.connect("mongodb://localhost:27017/", {
-        useUnifiedTopology: true,
-        useNewUrlParser: true,
-        // useCreateIndex: true,
-    });
-    server = app.listen();
-
-    // Creating user that we're testing on
-    const res = await request(app)
-        .post("/api/users")
-        .send({email: "nestor@mail.utoronto.ca", password: "abcd"})
-    expect(res.statusCode).toEqual(201);
-    expect(res.body.email).toEqual("nestor@mail.utoronto.ca");
-    userId = res.body._id;
+  await mongoose.connect("mongodb://localhost:27017/", {
+    useUnifiedTopology: true,
+    useNewUrlParser: true,
+    // useCreateIndex: true,
+  });
+  server = app.listen();
+
+  // Creating user that we're testing on
+  const res = await request(app)
+    .post("/api/users")
+    .send({ email: "nestor@mail.utoronto.ca", password: "abcd" });
+  expect(res.statusCode).toEqual(201);
+  expect(res.body.email).toEqual("nestor@mail.utoronto.ca");
+  userId = res.body._id;
 });
 
 describe("Example Test Suite", () => {
-    it("Root url sample test", async () => {
-        const res = await request(app).get("/")
-        expect(res.statusCode).toEqual(200);
-    })
-})
+  it("Root url sample test", async () => {
+    const res = await request(app).get("/");
+    expect(res.statusCode).toEqual(200);
+  });
+});
 
 describe("Login Test Suite", () => {
-    it("Valid Login", async () => {
-        const res = await request(app)
-            .post("/api/users/login")
-            .send({email: "nestor@mail.utoronto.ca", password: "abcd"})
-        expect(res.statusCode).toEqual(200)
-        expect(res.body).toHaveProperty("token")
-        jwt = res.body.token;
-
-    })
-
-    it("Invalid Login", async () => {
-        const res = await request(app)
-            .post("/api/users/login")
-            .send({email: "nestor@mail.utoronto.ca", password: ""})
-        expect(res.statusCode).toEqual(401);
-        expect(res.body).toHaveProperty("message");
-        expect(res.body.message).toEqual("Wrong password")
-    })
-})
+  it("Valid Login", async () => {
+    const res = await request(app)
+      .post("/api/users/login")
+      .send({ email: "nestor@mail.utoronto.ca", password: "abcd" });
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toHaveProperty("token");
+    jwt = res.body.token;
+  });
+
+  it("Invalid Login", async () => {
+    const res = await request(app)
+      .post("/api/users/login")
+      .send({ email: "nestor@mail.utoronto.ca", password: "" });
+    expect(res.statusCode).toEqual(401);
+    expect(res.body).toHaveProperty("message");
+    expect(res.body.message).toEqual("Wrong password");
+  });
+});
+
+describe("Getting Tasks Suite", () => {
```

Comment:
I would also add some additional tests such that they cover the different scenarios of date ranges (the 3 ranges that we searched for with the OR statement in Mongoose).

Reply:
For the comment above, there should at least be 3 test cases (1 for each scenario)